src/components/AssignTeamTableCell/index.tsx

```diff
@@ -0,0 +1,163 @@
+import { keys } from 'lodash';
+import React, { MouseEvent, useState } from 'react';
+import { connect } from 'react-redux';
+import { Button, Form, Popover, PopoverBody, PopoverHeader } from 'reactstrap';
+import { Store } from 'redux';
+import { ASSIGN_TEAMS, TEAMS_ASSIGNED } from '../../constants';
+import OrganizationSelect from '../../containers/forms/OrganizationSelect';
+import { stopPropagationAndPreventDefault } from '../../helpers/utils';
+import assignmentReducer, {
+  Assignment,
+  fetchAssignments,
+  getAssignmentsArrayByPlanIdByJurisdictionId,
+  reducerName as assignmentReducerName,
+} from '../../store/ducks/opensrp/assignments';
+import organizationsReducer, {
+  getOrganizationsById,
+  Organization,
+  reducerName as organizationsReducerName,
+} from '../../store/ducks/opensrp/organizations';
+
+reducerRegistry.register(assignmentReducerName, assignmentReducer);
+
+reducerRegistry.register(organizationsReducerName, organizationsReducer);
+
+/** Interface for Assign Teams cell props */
+export interface AssignTeamCellProps {
+  assignments: Assignment[];
+  fetchAssignmentsActionCreator: typeof fetchAssignments;
+  jurisdictionId: string;
+  organizationsById: { [key: string]: Organization } | null;
+  planId: string;
+}
+
+/** Component that will be rendered in IRS planning table cells
+ * showing the button and card in the Assign Teams column
+ */
+const AssignTeamTableCell = (props: AssignTeamCellProps) => {
+  const {
+    assignments,
+    fetchAssignmentsActionCreator,
+    jurisdictionId,
+    organizationsById,
+    planId,
+  } = props;
+  const [isActive, setIsActive] = useState<boolean>(false);
+
+  const onPlanAssignmentButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const onClearAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    const nextAssignments = assignments.filter(
+      (a: Assignment) => a.jurisdiction !== jurisdictionId
+    );
+    fetchAssignmentsActionCreator(nextAssignments);
+  };
+
+  const onSaveAssignmentsButtonClick = (e: MouseEvent) => {
+    stopPropagationAndPreventDefault(e);
+    setIsActive(!isActive);
+  };
+
+  const AssignTeamButton = (
+    <Button
+      color="primary"
+      id={getButtonId(jurisdictionId)}
+      onClick={onPlanAssignmentButtonClick}
+      size="sm"
+    >
+      {ASSIGN_TEAMS}
+    </Button>
+  );
+
+  const organizationsArray: Organization[] | null =
+    organizationsById &&
+    keys(organizationsById)
+      .map((o: string) => organizationsById[o])
+      .filter(o => !!o);
+
+  const organizationSelectProps = {
+    jurisdictionId,
+    name: getFormName(jurisdictionId),
+    planId,
+  };
+
+  const popoverProps = {
+    isOpen: isActive,
+    onClick: (e: React.MouseEvent) => stopPropagationAndPreventDefault(e),
+    target: getButtonId(jurisdictionId),
+    toggle: () => setIsActive(!isActive),
+  };
+
+  const AssignPopover = (
```

Comment:
Is it possible to make this a component that is passed into AssignTeamTableCell as a prop?

Reply:
@moshthepitt done! As a default the `AssignTeamTableCell` will fall back to the `AssignTeamPopover`